mne/minimum_norm/resolution_metrics.py

```diff
@@ -0,0 +1,397 @@
+
+Resolution metrics: localisation error, spatial extent, relative amplitude.
+Metrics can be computed for point-spread and cross-talk functions (PSFs/CTFs).
+"""
+import numpy as np
+
+from mne import SourceEstimate
+from mne.utils import logger, verbose
+
+
+@verbose
+def resolution_metrics(resmat, src, function, kind, metric):
+    """Compute spatial resolution metrics for linear solvers.
+
+    Parameters
+    ----------
+    resmat : array, shape (n_orient*n_locations, n_locations)
+        The resolution matrix.
+        If not a square matrix and if the number of rows is a multiple of
+        number of columns (e.g. free or loose orientations), then the Euclidean
+        length per source location is computed (e.g. if inverse operator with
+        free orientations was applied to foward solution with fixed
+        orientations).
+    src : instance of SourceSpaces
+        Source space object from forward or inverse operator.
+    function: str
+        'psf'|'ctf'
+        Whether to compute metrics for columns (point-spread functions, PSFs)
+        or rows (cross-talk functions, CTFs) of the resolution matrix.
+    kind : str
+        What kind of resolution feature to consider.
+        Options are 'localization_error' | 'spatial_extent' | 'amplitude'.
+    metric : str
+        The exact resolution metric for the kind of resolution feature.
+        This must match with 'kind'. The allowed options are:
+        kind='localization_error':
+            'peak': Peak localization error (PLE), Euclidean distance between
+                    peak
+                    and true source location.
+            'cog': Centre-of-gravity localisation error (CoG), Euclidean
+                   distance between CoG and true source location.
+        kind='spatial_extent':
+            'sd': spatial deviation (e.g. Molins et al.).
+            'maxrad': maximum radius to 50% of max amplitude.
+        kind='relative_amplitude':
+            'peak': Ratio between absolute maximum amplitudes of peaks per
+                    location and maximum peak across locations.
+            'sum': Ratio between sums of absolute amplitudes.
+
+    Returns
+    -------
+    resolution_metric : instance of SourceEstimate
+        The source estimate contains the resolution metric as an array with
+        shape (n_locations,)
+    """
+    # Check if input options are valid
+    if kind == 'localization_error':
+        if metric not in ['peak', 'cog']:
+            print('Not an allowed metric for localization_error: %s' % metric)
+
+    elif kind == 'spatial_extent':
+        if metric not in ['sd', 'maxrad']:
+            print('Not an allowed metric for spatial_extent: %s' % metric)
+
+    elif kind == 'relative_amplitude':
+        if metric not in ['peak', 'sum']:
+            print('Not an allowed metric for amplitude: %s' % metric)
+
+    else:
+        print('Not a recognised kind of resolution feature: %s.' % kind)
+
+        return
```

Comment:
use a ValueError exception rather than a return

Reply:
This means to replace "print(" with "raise Value Error(", and simply remove the "return"?